tools/harness-automation/parse_topofile.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2020, The OpenThread Authors.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the
+#    names of its contributors may be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+"""
+parse_topofile.py
+-----------------
+This script is used to parse TopologyConfig.txt file to list vendor device info
+when preparing for the Thread Certification testbed
+
+usage: parse_topofile.py [-h] [-f TOPO_FILE] [-c CASE_LIST [CASE_LIST ...]]
+
+    parse TopologyConfig file and list all devices by case
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      -f TOPO_FILE          Topology config file (default: C:/GRL/Thread1.1/Thread
+                            _Harness/TestScripts/TopologyConfig.txt)
+      -c CASE_LIST [CASE_LIST ...]
+                            Test case list (e.g. 5.1.1 9.2.1, default: all)
+
+Examples:
+    1. Get case 5.1.1 vendor info
+       cmd:   python parse_topofile.py -f TopologyConfig.txt -c 5.1.1
+       output:
+       case 5.1.1:
+            role-vendor pair: [('Leader', 'ARM'), ('Router_1', 'ARM')]
+            needed vendor devices:{'ARM': 2}
+
+       Testbed needed vendor devices:{'ARM': 2}
+
+    2. Get case 5.1.1 and 5.2.1 vendor info
```

Comment:
add example to list all required reference devices? I think it would be a scenario mostly used.